apps/full-site-editing/bin/npm-run-build.js

```diff
@@ -26,6 +27,20 @@ const runOptions = {
 	printLabel: true,
 };
 
-runAll( commands, runOptions ).then( () => {
-	console.log( 'Finished running commands!' );
-} );
+runAll( commands, runOptions )
+	.then( () => {
+		console.log( 'Finished running commands!' );
+	} )
+	.catch( ( e ) => {
+		console.log( 'The build errored.' );
+		const tasks = e.results;
+		if ( Array.isArray( tasks ) ) {
+			const didNewspackSyncFail = tasks.some(
+				( task ) => task.name === 'build:newspack-blocks' && task.code !== 0
+			);
+			if ( didNewspackSyncFail ) {
+				console.log( 'You may need to run `composer install` from wp-calypso root.' );
+			}
+		}
+		process.exit( 1 );
```

Comment:
without process.exit, the build command simply throws an error. Unfortunately, in NodeJS, an uncaught error does not yet result in exiting with a non-zero code. So we need to exit manually.

Reply:
We probably want `process.exitCode = 1;` here, rather than `process.exit()`, particularly with `parallel: true`.

Reply:
done